usecases/no2-monitoring/index.md

```diff
@@ -0,0 +1,384 @@
+```
+
+</template>
+<template v-slot:r>
+
+```r
+p = processes()
+
+year = "2019"
+extent = list( # Münster
+  "type" = "Polygon",
+  "coordinates" = list(list(
+    list(5.9521131313239675, 51.78636868072434),
+    list(5.971674074614773, 50.52897921775278),
+    list(9.893643204421315, 50.609735254808925),
+    list(9.864301789485104, 51.870991335721925),
+    list(5.9521131313239675, 51.78636868072434)
+  ))
+)
+datacube = p$load_collection(
+  id = "TERRASCOPE_S5P_L3_NO2_TD_V1",
+  spatial_extent = extent,
+  temporal_extent = list(paste(year, "-01-01", sep = ""), paste(year, "-12-31", sep = ""))
+)
+```
+
+</template>
+</CodeSwitcher>
+
+### 2. Fill gaps
+
+The data cube may contain no-data values due to the removal of clouds in the pre-processing of the collection.
+We'll apply a linear interpolation along the temporal dimension:
+
+<CodeSwitcher :languages="{py: 'Python', js: 'JavaScript', r: 'R'}">
+<template v-slot:py>
+
+```python
+datacube = datacube.apply_dimension(dimension = "t", process = "array_interpolate_linear")
+```
+
+</template>
+<template v-slot:js>
+
+```js
+let interpolate = (data, _, builder) => builder.array_interpolate_linear(data);
+datacube = builder.apply_dimension(datacube, interpolate, "t");
+```
+
+</template>
+<template v-slot:r>
+
+```r
+interpolate = function(data, context) {
+  p$array_interpolate_linear(data = data)
+}
+datacube = p$apply_dimension(data = datacube, dimension = "t", process = interpolate)
+```
+
+</template>
+</CodeSwitcher>
+
+### 3. Smoothen values (optional)
+
+If you want to smoothen the values to get rid of noise for example, we can run a moving average over a certain amount of days over the temporal dimension.
+If you want to work on the raw values, you can also omit this step.
+The `moving_average_window` variable specifies the smoothing in number of days.
+You can choose it freely, but it needs to be an odd integer >= 3.
+In the example below 31 was chosen to smooth the timeseries with a moving average of a full month.
+
+<CodeSwitcher :languages="{py: 'Python', js: 'JavaScript', r: 'R'}">
+<template v-slot:py>
+
+```python
+moving_average_window = 31
+udf = openeo.UDF("""
+from pandas import Series
+import numpy as np
+
+def apply_timeseries(series: Series, context: dict) -> Series:
+    return np.convolve(series, np.ones({n})/{n}, mode='same')
+""".format(n = moving_average_window))
+datacube = datacube.apply_dimension(dimension = "t", process = udf)
+```
+
+</template>
+<template v-slot:js>
+
+```js
+let moving_average_window = 31;
+let udf = `
+from pandas import Series
+import numpy as np
+
+def apply_timeseries(series: Series, context: dict) -> Series:
+    return np.convolve(series, np.ones(${moving_average_window})/${moving_average_window}, mode='same')
+`;
+let run = (data, _, builder) => builder.run_udf(data, udf, "Python");
+datacube = builder.apply_dimension(datacube, run, "t")
+```
+
+</template>
+<template v-slot:r>
+
+```r
```

Comment:
This is ugly, but I guess our only chance atm?
Would it be possible to note somewhere that this is an intermediate solution?

Reply:
There's no R UDF here?!
This is a pure Python UDF for Platform. The alternative is storing it in a file and that is mentioned in the text.

Reply:
Ok now I see what's happening: A python UDF run through the R-Client.